tooling/release_download_test.sh

```diff
@@ -8,164 +8,470 @@
 #   1 - Something fundamentally wrong before we could check anything
 #   2 - GPG signature verification failed
 #   3 - SHA checksum failed
-#   4 - aarch64 detected GCC/GLIBC version not as expected
+#   4 - detected GCC/GLIBC version not as expected
 #   5 - CylconeDX validation checks failed
 #   6 - SBOM contents did not meet expectations
 # Note that if there are multiple failures the highest will be the exit code
 # If there is a non-zero exit code check the output for "ERROR:"
 # 
 # For future enhancement ideas, see https://github.com/adoptium/temurin-build/issues/3506#issuecomment-1783237963
 #
-STARTDIR="$PWD"
-TAG=${1:-$TAG}
-[ -z "$TAG" ] && echo "Usage: $0 TAG" && exit 1
-[ "$(uname -m)" != "aarch64" ] && echo "This script is hard coded to be run on Linux/aarch64 - aborting" && exit 1
-
-[ "$VERBOSE" = "true" ] && set +x
-if echo "$TAG" | grep jdk8u; then
-  MAJOR_VERSION=8
-elif echo "$TAG" | grep ^jdk-; then
-  MAJOR_VERSION=$(echo "$TAG" | cut -d- -f2 | cut -d. -f1 | cut -d\+ -f1)
-else
-  # Probably a beta with the tag starting jdkXXu
-  MAJOR_VERSION=$(echo "$TAG" | cut -d- -f1 | tr -d jdku)
-fi
 
-echo "$(date +%T) : IVT : I will be checking https://github.com/adoptium/temurin${MAJOR_VERSION}-binaries/releases/tag/$TAG"
-if [ -z "${MAJOR_VERSION}" ] || [ -z "${TAG}" ]; then
-   echo "MAJOR_VERSION or TAG undefined - aborting"
-   exit 1
-fi
+set -euo pipefail
 
-if ! curl -sS "https://api.github.com/repos/adoptium/temurin${MAJOR_VERSION}-binaries/releases" > "$WORKSPACE/jdk${MAJOR_VERSION}.txt"; then
-   echo "github API call failed - aborting"
-   exit 2
-fi
+WORKSPACE=${WORKSPACE:-"$PWD"}
+VERBOSE=false
+KEEP_STAGING=false
+SKIP_DOWNLOADING=false
+
+TAG=""
+MAJOR_VERSION=""
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# shellcheck source=sbin/common/config_init.sh
+source "$SCRIPT_DIR/common_logging.sh"
+
+usage() {
```

Comment:
Noting that this will require a change to the calling jenkins job when merged as the script is currently parameterless

Reply:
I am not super happy with this include here, the common_logging.sh will require that a parameter VERBOSE is defined, do you have any suggestion here?

Reply:
Not sure I understand your comment here - you could pass a verbose option into `common_logging.sh` if required, but I feel I've missed the point :-)

I have no problems with the `-v` flag as-is unless you see an issue with it, but my comment was related to the fact that the mandatory `TAG` parameter is new (again, not a problem, and it's a good change, just needs dealing with in the automation when we merge this)

Reply:
before that, the script did have this line

TAG=${1:-$TAG}

which I did not fully understand, does it mean that TAG will be assigned the first parameter. If this is absent, it will use the TAG variable that is defined externally e.g. using export before running the script? If this is needed for running on jenkins, we can readd it ofc.

Reply:
I reverted to the original behavior such that the TAG can be passed in as argument and as env variable.

Reply:
> does it mean that TAG will be assigned the first parameter. If this is absent, it will use the TAG variable that is defined externally e.g. using export before running the script?

Yes that's correct. I'm ok with either approach and either will work in jenkins, would just need a one line modification in the jenkins script, but sine you've reverted it I'm happy with that :-)